### ISSUE TITLE
Implement Exciting-Mixing, Scalar and Diagonal Broyden approximations

### ISSUE THREAD

**Issue by ChrisRackauckas** [SciML/SimpleNonlinearSolve.jl#33]
https://docs.scipy.org/doc/scipy/reference/optimize.root-excitingmixing.html
https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.diagbroyden.html#scipy.optimize.diagbroyden
https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.linearmixing.html#scipy.optimize.linearmixing

**Comment by ChrisRackauckas** [comment]
@avik-pal  is this completed as well?

**Comment by avik-pal** [comment]
No. Maybe let's transfer this to NonlinearSolve? We can handle all of them with branches in Broyden there
